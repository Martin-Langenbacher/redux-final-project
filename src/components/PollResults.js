const PollResults = ({ optionOneAmount, optionTwoAmount }) => {
  const totalAmount = optionOneAmount + optionTwoAmount;

  

  const percentOne = calculatePercentage(optionOneAmount, totalAmount);
  const percentTwo = calculatePercentage(optionTwoAmount, totalAmount);
  // const percentOne = Math.round(1000 * (optionOneAmount / totalAmount)) / 10;
  // const percentTwo = Math.round(1000 * (optionTwoAmount / totalAmount)) / 10;

  /*
  module.exports = {
    calculatePercentage,
  };
  */

  return (
    <>
      <div className="poll-result-distance"></div>

      <div className="line"></div>
      <div className="center">
        <h2 className="poll-result-title">Poll Results:</h2>
      </div>
      <div>
        <div className="center">
          <div className="container-poll-page">
            <div className="poll-text-left">
              <div className="poll-result">{optionOneAmount}</div>
            </div>
            <div className="poll-text-right">
              <div className="poll-result">{optionTwoAmount}</div>
            </div>
          </div>
        </div>
        <div className="center">
          <div className="container-poll-page">
            <div className="poll-text-left">
              <div className="poll-result">{percentOne}%</div>
            </div>
            <div className="poll-text-right">
              <div className="poll-result">{percentTwo}%</div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export function  calculatePercentage (amount, totalAmount) {
  return Math.round(1000 * (amount / totalAmount)) / 10;
};

export default PollResults;
