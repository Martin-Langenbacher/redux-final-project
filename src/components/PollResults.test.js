// INFORMATION
// press 'u', if the view really changed ! --> You get a new snapshot!

// is present:  ==> getBy
// not present: ==> queryBy

import * as React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import PollResults from "./PollResults";

const mockStore = configureStore();

describe("PollResults", () => {
  it("10a) shows, that the polls fit on percentage of people and display it corecctly (Snapshot)", () => {
    const optionOneAmount = 2;
    const optionTwoAmount = 1;
    const store = mockStore({});
    var view = render(
      <Provider store={store}>
        <Router>
          <PollResults
            optionOneAmount={optionOneAmount}
            optionTwoAmount={optionTwoAmount}
          />
        </Router>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
  

  it("10b) shows, that the polls fit on percentage of people (%-numbers correct)", () => {
    const optionOneAmount = 7;
    const optionTwoAmount = 9;
    const store = mockStore({});
    var view = render(
      <Provider store={store}>
        <Router>
          <PollResults
            optionOneAmount={optionOneAmount}
            optionTwoAmount={optionTwoAmount}
          />
        </Router>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });




});
