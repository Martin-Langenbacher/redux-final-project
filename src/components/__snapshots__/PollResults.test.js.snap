// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollResults 10a) shows, that the polls fit on percentage of people and display it corecctly (Snapshot) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="poll-result-distance"
      />
      <div
        class="line"
      />
      <div
        class="center"
      >
        <h2
          class="poll-result-title"
        >
          Poll Results:
        </h2>
      </div>
      <div>
        <div
          class="center"
        >
          <div
            class="container-poll-page"
          >
            <div
              class="poll-text-left"
            >
              <div
                class="poll-result"
              >
                2
              </div>
            </div>
            <div
              class="poll-text-right"
            >
              <div
                class="poll-result"
              >
                1
              </div>
            </div>
          </div>
        </div>
        <div
          class="center"
        >
          <div
            class="container-poll-page"
          >
            <div
              class="poll-text-left"
            >
              <div
                class="poll-result"
              >
                66.7
                %
              </div>
            </div>
            <div
              class="poll-text-right"
            >
              <div
                class="poll-result"
              >
                33.3
                %
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="poll-result-distance"
    />
    <div
      class="line"
    />
    <div
      class="center"
    >
      <h2
        class="poll-result-title"
      >
        Poll Results:
      </h2>
    </div>
    <div>
      <div
        class="center"
      >
        <div
          class="container-poll-page"
        >
          <div
            class="poll-text-left"
          >
            <div
              class="poll-result"
            >
              2
            </div>
          </div>
          <div
            class="poll-text-right"
          >
            <div
              class="poll-result"
            >
              1
            </div>
          </div>
        </div>
      </div>
      <div
        class="center"
      >
        <div
          class="container-poll-page"
        >
          <div
            class="poll-text-left"
          >
            <div
              class="poll-result"
            >
              66.7
              %
            </div>
          </div>
          <div
            class="poll-text-right"
          >
            <div
              class="poll-result"
            >
              33.3
              %
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PollResults 10b) shows, that the polls fit on percentage of people (%-numbers correct) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="poll-result-distance"
      />
      <div
        class="line"
      />
      <div
        class="center"
      >
        <h2
          class="poll-result-title"
        >
          Poll Results:
        </h2>
      </div>
      <div>
        <div
          class="center"
        >
          <div
            class="container-poll-page"
          >
            <div
              class="poll-text-left"
            >
              <div
                class="poll-result"
              >
                7
              </div>
            </div>
            <div
              class="poll-text-right"
            >
              <div
                class="poll-result"
              >
                9
              </div>
            </div>
          </div>
        </div>
        <div
          class="center"
        >
          <div
            class="container-poll-page"
          >
            <div
              class="poll-text-left"
            >
              <div
                class="poll-result"
              >
                43.8
                %
              </div>
            </div>
            <div
              class="poll-text-right"
            >
              <div
                class="poll-result"
              >
                56.3
                %
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="poll-result-distance"
    />
    <div
      class="line"
    />
    <div
      class="center"
    >
      <h2
        class="poll-result-title"
      >
        Poll Results:
      </h2>
    </div>
    <div>
      <div
        class="center"
      >
        <div
          class="container-poll-page"
        >
          <div
            class="poll-text-left"
          >
            <div
              class="poll-result"
            >
              7
            </div>
          </div>
          <div
            class="poll-text-right"
          >
            <div
              class="poll-result"
            >
              9
            </div>
          </div>
        </div>
      </div>
      <div
        class="center"
      >
        <div
          class="container-poll-page"
        >
          <div
            class="poll-text-left"
          >
            <div
              class="poll-result"
            >
              43.8
              %
            </div>
          </div>
          <div
            class="poll-text-right"
          >
            <div
              class="poll-result"
            >
              56.3
              %
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
